InnerEye/ML/run_ml.py

```diff
@@ -138,50 +154,112 @@ def __init__(self,
+        # parameters of the container will be copied into the module.
+        if self.output_subfolder:
+            # This codepath is only executed for cross validation runs outside AzureML: The folder structure
+            # uses an existing folder structure set by the caller, and just a subfolder is added.
+            self.container.file_system_config = self.container.file_system_config.add_subfolder(self.output_subfolder)
+        else:
+            self.container.create_filesystem(self.project_root)
+        # A lot of the code for the built-in InnerEye models expects the output paths directly in the config files.
+        if isinstance(self.container, InnerEyeContainer):
+            self.container.config.local_dataset = self.container.local_dataset
+            self.container.config.file_system_config = self.container.file_system_config
+        self.container.setup()
+        self.container.create_lightning_module_and_store()
+        self._has_setup_run = True
+
+    @property
+    def is_offline_run(self) -> bool:
+        """
+        Returns True if the present run is outside of AzureML, and False if it is inside of AzureML.
+        :return:
+        """
+        return is_offline_run_context(RUN_CONTEXT)
+
+    @property
+    def innereye_config(self) -> DeepLearningConfig:
+        """
+        Gets the model configuration object for all built-in InnerEye models. Raises an exception if the present
+        object trains a LightningContainer that is not a built-in InnerEye model.
+        """
+        if self.model_config is None or not isinstance(self.model_config, DeepLearningConfig):
+            raise ValueError("This property should only be used with built-in InnerEye models, but model "
+                             f"configuration is of type {type(self.model_config)}")
+        return self.model_config
+
+    def start_logging_to_file(self) -> None:
+        if self.container is None:
+            self.setup()
+        logging_to_file(self.container.logs_folder / LOG_FILE_NAME)
 
     def is_offline_cross_val_parent_run(self) -> bool:
         """
         Returns true if the current run is an offline run with cross validation splits > 0
         and cross_validation_split_index == DEFAULT_CROSS_VALIDATION_SPLIT_INDEX (ie: a parent)
         """
-        return self.model_config.cross_validation_split_index == DEFAULT_CROSS_VALIDATION_SPLIT_INDEX and \
-               self.model_config.perform_cross_validation and self.model_config.is_offline_run
+        return self.container.cross_validation_split_index == DEFAULT_CROSS_VALIDATION_SPLIT_INDEX and \
+               self.container.perform_cross_validation and self.is_offline_run
 
     def spawn_offline_cross_val_classification_child_runs(self) -> None:
         """
         Trains and Tests k models based on their respective data splits sequentially.
         Stores the results on the Validation set to the outputs directory of the parent run.
         """
```

Comment:
Should we add in the doc "Only for InnerEye built-in models"?

Reply:
Are we throwing an error if somebody does number_cross_validation > 1 for a bring your own module at the moment?

Reply:
I've not made the doc change, but added an error in `runner.py` as you suggested.